### DISCUSSION TITLE
Difference between Breeding value (bv); estimation of the breeding value (ebv); Genetic value (gv) in AlphasimR simulation

### DISCUSSION THREAD

**Discussion by Dkplucas** [gaynorr/AlphaSimR#112] [Q&A] [answered]
I'm using AlphasimR for my reseach work.
Actually, I tried to select a subset of nInd individuals from a population by the functun selectInd( ) with the argument use (the Breeding value (bv); estimation of the breeding value (ebv); genetic value (gv) and the phenotype (pheno). 
For the selection base on the phenotype it was easy to understand.
But for the selection based on the arguments bv; ebv; and gv. I had some difficulties to explain it. I want to know what is the equation used by AlphasiomR to calculate gv; bv and ebv?
According to the other interventions on the site, I read that Breeding value calculation was related to the Dominance but I would like to know then what is the difference between the three arguments bv; gv and ebv calculation in AlphasimR?
Here is an example of code to find it out, where i try to bring out the three values for a comparison. There were all different.

#########################################

#Trial for Estimated Breeding Value
rm(list =ls())
library(AlphaSimR)
#Create founder haplotypes
founderPop = quickHaplo(nInd=10, nChr=1, segSites=10)
#set simulation parameters
SP = SimParam$new(founderPop)
SP$addTraitAD(10, meanDD=0.5)
SP$setVarE(h2=0.5)
#Create population
pop = newPop(founderPop,SimParam = SP)
pop@ebv = matrix(rnorm(pop@nInd), nrow=pop@nInd, ncol=1)
ebv(pop)
gv(pop)
bv(pop, simParam= SP)
write.csv(ebv(pop), file = "Estimation of the breeding value.csv")
write.csv(gv(pop), file = "Genetic Value.csv")
write.csv(bv(pop, simParam= SP), file = "Breeding Value.csv")

########################################

![image_2023-02-17_124953351](https://user-images.githubusercontent.com/110559557/219544852-668f128d-c90f-4c09-9113-ebfde96ea51a.png)

**Comment by gregorgorjanc** [comment, marked as answer]
@Dkplucas genetic value represents the value of all DNA in an individual, which includes all additive effects of alleles, dominance effects of pairs of alleles at each locus, and epistatic effects of all pairs (or more) of alleles across loci.

Breeding value represents the value of DNA that individual passes to the next generation, which is not genetic value, since an individual does not pass all of DNA but just a particular set due to recombination and segregation.

The above two are true values, which we have access to in a simulation.

Estimate of breeding value (EBV) is obtained from some phenotype data. In your case, you have  generated some random numbers as EBV irrespective of the true breeding value of an individual, so you can expect accuracy of these estimates to be zero. You obviously want a better accuracy so you will need some estimation procedure, possibly from an R package. AlphaSimR has some genomic selection models implemented though if that's what you are after.

To study the difference between genetic and breed value I suggest the green book from Falconer and MacKay: Introduction to quantitative genetics.

To study estimation of breeding values I suggest the blue book from Mrode: Linear models for the Estimation of breeding values (Something like this).

**Comment by Dkplucas** [comment]
Thanks so much Professor, it is very helpful. For a better accuracy of the EBV, please which R package can I use?

**Reply by gregorgorjanc** [reply]
AlphaSimR has some functionality for genomic prediction models. There are also other R packages, for example pedigree, pedigreemm, BGLR, MCMCglmm, sommer, asreml, and others. There are also specialised standalone software such as blupf90, WOMBAT, Mix99, ASReml, etc.

**Comment by Dkplucas** [comment]
Thank so much Professor
